### ISSUE TITLE
Pen movement on canvas area not working

### ISSUE THREAD

**Issue by Zacrain** [mbrlabs/Lorien#303]
**Lorien version:**
Current `main` branch. Commit hash: [0bd6fff](https://github.com/mbrlabs/Lorien/commit/0bd6fff554cc79ea2457c4371168ffeaaa6f764e)

**OS/device including version:**
Windows 10
Wacom Intuos BT M CTL-6100WL, Driver version: `6.4.7-3`

**Issue description:**
The movement of the graphic tablet's pen is not reflected on the canvas. I can draw a point where the cursor is currently at, but I can't move it with the pen. There are no issues with my mouse though. Clicking buttons like selecting tools works using the pen as long as I moved the cursor into the top bar before with my mouse (or with the pen if I moved it out of the window and move it back again).

**Steps to reproduce:**
- Connect the graphic tablet.
- Run Lorien (either as release build or debug through Godot, doesn't matter).
- Move cursor on the canvas using the mouse.
- Try to move the cursor using the pen now.

**Further Observations:**
The positional information is correctly received at all times when using the pen. I've tested this by simply adding a print at line `80` in `InfiniteCanvas.gd`:
```gdscript
func _process_event(event: InputEvent) -> void:
	if event is InputEventMouseMotion:
		print("Mouse position at: ", event.position)
		info.current_pressure = event.pressure
                #...
```

However, there seems to be a problem with the tool's cursor on the canvas in some way which leads to it not being updated correctly when using a pen. Which I find curious as pen movements are usually handled simply as mouse movements in Godot, if I'm not mistaken. Therefore I wonder whether there is something specific to the project which is causing this issue.

Recognition of applied pressure seems to be working.

**Comment by Zacrain** [comment]
This seems to be an issue related to how Godot 4.x processes pen inputs. As a workaround it was suggested to switch from absolute to relative positioning in the graphic tablet's driver settings. Godot currently does not seem to be able to process absolute positioning of pens sufficiently.

See:
https://github.com/godotengine/godot/issues/76451
https://github.com/godotengine/godot/issues/75903

So in my Wacom tablet settings setting to "mouse mode" fixed this for now:
![image](https://github.com/user-attachments/assets/de80ad49-3ffc-46ca-9916-3b54c766495d)

However, the pen works in pen mode (absolute positioning) completely fine in Lorien's toolbar. Just not on the canvas. 🤔

**Comment by mbrlabs** [comment]
That's odd, thanks for the report! Does this only happen on the latest main branch? Did you test this with v0.6.0 as well?

I'm on Linux and here both absolute and relative positioning work fine with Wacoms.

**Comment by mzpsh** [comment]
Confirming this on windows with huion h420 with latest build from https://github.com/mbrlabs/Lorien/actions/runs/10863437411.

**Comment by Zacrain** [comment]
> That's odd, thanks for the report! Does this only happen on the latest main branch? Did you test this with v0.6.0 as well?
> 
> I'm on Linux and here both absolute and relative positioning work fine with Wacoms.

It doesn't happen on v0.6.0, which suggests that it might indeed be a problem stemming from Godot 4.x.

**Comment by mbrlabs** [comment]
Alright, can confirm as well on my old Windows 10 box. I'll look into it!

**Comment by mbrlabs** [comment]
I think i fixed it or at least found a workaround. When changing the windows tablet driver (in the Godot project settings) to `wintab` it works. The default is `winink`.

If this works for you as well we should make it the default for Windows and provide an option in the settings to switch between all available drivers.

**Comment by mzpsh** [comment]
> I think i fixed it or at least found a workaround. When changing the Windows tablet driver (in Godot's advanced project settings: Input Devices -> Pen Tablet -> Driver) to `wintab` it works. The default is `winink`.
> 
> If this works for you as well we should make it the default for Windows and provide an option in the settings to switch between all available drivers.
> 
> CI builds for testing: https://github.com/mbrlabs/Lorien/actions/runs/10983725293

It still doesnt work on me with the latest build. FYI the pressure works, just like before.
Perhaps Godot's problem?

**Comment by mbrlabs** [comment]
Did you try the CI build from the PR? I did not merge this yet.

**Comment by mzpsh** [comment]
@mbrlabs Yes, I did try the build from the CI. Also tried that tablet_driver branch on godot, and tried to tweak tablet driver and emulate mouse from touch/touch from mouse, still no luck.

**Comment by mbrlabs** [comment]
Ok, thank you testing! At least it seems to work for Wacoms now so i'm merging the PR and hope it also works for @Zacrain.

Huion tablets like yours might need a different treatment...or a fix in Godot itself. By the way: are you on Windows 10 or 11?

**Comment by mzpsh** [comment]
Windows 10 19044, ltsc 2021.

**Comment by mbrlabs** [comment]
@mzpsh ok, so i got a Huion 420X to test this. I installed the latest drivers (v15.7.6.1314) on Windows 10 and for me the `wintab` driver in the Lorien settings works fine right out of the box. `winink` has the same issue as Wacom.

Maybe your Huion driver is outdated or your device is not supported anymore?

**Comment by mzpsh** [comment]
Tried H420X driver on H420 in windows, no luck. Winink can detect pressure, but cursor not moving. Wintab can move the cursor, but no pressure detection.
On Arch, it's fully functional. I think we can conclude the tablet just not supported anymore on Windows and it seems Godot pen tablet handling on Windows has problem?
